--- Firebird-5.0.0.1306-0-source/builds/install/misc/firebird.conf.orig	2024-01-11 11:24:21.000000000 +0100
+++ Firebird-5.0.0.1306-0-source/builds/install/misc/firebird.conf	2024-02-22 09:29:00.475211032 +0100
@@ -775,6 +775,7 @@
 #	Type: string
 #
 #DataTypeCompatibility =
+DataTypeCompatibility = 3.0
 
 
 # ===========================
@@ -1258,3 +1259,5 @@
 # Type: integer
 #
 #ExtConnPoolLifeTime = 7200
+
+include conf.d/*.conf
