--- firebird-1.5.0.4290/builds/posix/Makefile.in.extlib.orig	2004-02-15 09:51:02.000000000 +0100
+++ firebird-1.5.0.4290/builds/posix/Makefile.in.extlib	2004-05-24 23:58:14.519459120 +0200
@@ -110,7 +110,7 @@
 
 lib_ib_util: $(LIBIBUTIL_SO)
 
-$(LIBIBUTIL_SO):	LINK_PROD=$(LIBIBUTIL_SO)
+$(LIBIBUTIL_SO):	LINK_PROD=$(IbUtilLibraryName)
 $(LIBIBUTIL_SO):	$(UTIL_Objects)
 	$(LIB_LINK) $(LIB_LINK_OPTIONS) $(LIB_LINK_SONAME_LINE) $(LIB_LINK_RPATH_LINE) -o $@ $^
 #	-$(LN) $(@F) $(LIBFBINTL_SOBASENAME) 
