diff -uNr firebird-1.5.6.5026.orig/src/jrd/pag.cpp firebird-1.5.6.5026/src/jrd/pag.cpp
--- firebird-1.5.6.5026.orig/src/jrd/pag.cpp	2012-04-14 23:33:16.904679000 +0200
+++ firebird-1.5.6.5026/src/jrd/pag.cpp	2012-04-14 23:36:16.126129942 +0200
@@ -144,7 +144,10 @@
 static const int CLASS_DARWIN_PPC = 23;   // Darwin/PowerPC
 static const int CLASS_LINUX_AMD64 = 24;  // LINUX on AMD64 systems
 static const int CLASS_DARWIN_I386 = 25;  // Darwin/Intel
-static const int CLASS_MAX10 = CLASS_DARWIN_I386;
+static const int CLASS_FREEBSD_AMD64 = 26;// FreeBSD/amd64
+static const int CLASS_WINDOWS_AMD64 = 27;// Windows/amd64
+static const int CLASS_LINUX_PPC = 28;    // LINUX/PowerPC
+static const int CLASS_MAX10 = CLASS_LINUX_PPC;
 
 // ARCHITECTURE COMPATIBILITY CLASSES
 
@@ -186,10 +189,13 @@
 	archIntel86, // CLASS_LINUX_I386
 	archUnknown, // CLASS_LINUX_SPARC
 	archIntel86, // CLASS_FREEBSD_I386
-    archIntel86, // CLASS_NETBSD_I386
+	archIntel86, // CLASS_NETBSD_I386
 	archUnknown, // CLASS_DARWIN_PPC
 	archUnknown, // CLASS_LINUX_AMD64
-	archIntel86  // CLASS_DARWIN_I386
+	archIntel86, // CLASS_DARWIN_I386
+	archUnknown, // CLASS_FREEBSD_AMD64
+	archUnknown, // CLASS_WINDOWS_AMD64
+	archUnknown  // CLASS_LINUX_PPC
 };
 
 #ifdef sun
@@ -402,6 +408,9 @@
 	return TRUE;
 }
 
+#ifdef PPC
+#define CLASS		CLASS_LINUX_PPC
+#endif
 
 USHORT PAG_add_file(TEXT * file_name, SLONG start)
 {
