--- Firebird-3.0.10.33601-0/src/yvalve/gds.cpp.orig	2022-07-01 08:34:37.770832195 +0200
+++ Firebird-3.0.10.33601-0/src/yvalve/gds.cpp	2022-07-01 08:41:44.837560283 +0200
@@ -3826,10 +3826,7 @@
 		if (!fb_utils::readenv(FB_LOCK_ENV, lockPrefix))
 		{
 #if !defined(WIN_NT)
-			const char* tmp = getTemporaryFolder();
-			if (!tmp)
-				tmp = WORKFILE;
-			PathUtils::concatPath(lockPrefix, tmp, LOCKDIR);
+			PathUtils::concatPath(lockPrefix, "/var/lib", LOCKDIR);
 #else
 #ifdef WIN9X_SUPPORT
 			// shell32.dll version 5.0 and later supports SHGetFolderPath entry point
