--- Firebird-3.0.4.33054-0/builds/posix/prefix.linux.orig	2018-10-22 15:04:00.969062218 +0200
+++ Firebird-3.0.4.33054-0/builds/posix/prefix.linux	2018-10-31 07:52:26.717188011 +0100
@@ -18,9 +18,9 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
 CXXFLAGS=-std=gnu++03
-OPTIMIZE_FLAGS=-O3 -march=i586 -mtune=i686 -fno-omit-frame-pointer
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
diff -up Firebird-3.0.9.33560-0/builds/posix/prefix.linux_amd64.orig Firebird-3.0.9.33560-0/builds/posix/prefix.linux_amd64
--- Firebird-3.0.9.33560-0/builds/posix/prefix.linux_amd64.orig	2022-04-25 12:12:11.720212530 +0200
+++ Firebird-3.0.9.33560-0/builds/posix/prefix.linux_amd64	2022-04-25 13:04:31.045965585 +0200
@@ -18,9 +18,9 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup

-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
 CXXFLAGS=-std=gnu++03
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-invalid-offsetof -Wno-narrowing -Wno-unused-local-typedefs -Wno-class-memaccess

 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
diff -up Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm.orig Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm
--- Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm.orig	2022-04-25 12:12:11.720212530 +0200
+++ Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm	2022-04-25 13:07:08.771586881 +0200
@@ -21,9 +21,9 @@
 #PROD_FLAGS=-O3 -DLINUX -DARM -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
 #DEV_FLAGS=-ggdb -DLINUX -DARM -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor -DFB_SEND_FLAGS=MSG_NOSIGNAL

-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM -pipe -p -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks -fsigned-char
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM -pipe -p -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks -fsigned-char
 CXXFLAGS=-std=gnu++03
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-invalid-offsetof -Wno-narrowing -Wno-unused-local-typedefs -Wno-non-virtual-dtor

 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
diff -up Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm64.orig Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm64
--- Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm64.orig	2022-04-25 12:12:11.720212530 +0200
+++ Firebird-3.0.9.33560-0/builds/posix/prefix.linux_arm64	2022-04-25 13:07:58.594590193 +0200
@@ -21,9 +21,9 @@
 #PROD_FLAGS=-O3 -DLINUX -DARM64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
 #DEV_FLAGS=-ggdb -DLINUX -DARM64 -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor -DFB_SEND_FLAGS=MSG_NO

-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM64 -pipe -p -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks -fsigned-char
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM64 -pipe -p -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks -fsigned-char
 CXXFLAGS=-std=gnu++03
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-invalid-offsetof -Wno-narrowing -Wno-unused-local-typedefs -Wno-non-virtual-dtor

 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-3.0.4.33054-0/builds/posix/prefix.linux_generic.orig	2018-10-22 15:04:00.969062218 +0200
+++ Firebird-3.0.4.33054-0/builds/posix/prefix.linux_generic	2018-10-31 08:25:12.372864962 +0100
@@ -18,8 +18,9 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL -fno-delete-null-pointer-checks
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL -fno-delete-null-pointer-checks
 CXXFLAGS=-std=gnu++03
+OPTIMIZE_FLAGS=
 
-PROD_FLAGS=-ggdb -O3 $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Wall -Wno-switch $(COMMON_FLAGS) -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=-Wall -Wno-switch $(COMMON_FLAGS) -Wno-non-virtual-dtor
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_ia64.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_ia64	2016-08-31 20:55:10.367740946 +0200
@@ -18,8 +18,8 @@
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 # Adriano dos Santos Fernandes - IA-64 port
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DIA64 -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer 
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DIA64 -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer 
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_mips.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_mips	2016-08-31 20:56:05.504406983 +0200
@@ -1,5 +1,5 @@
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_mipsel.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_mipsel	2016-08-31 21:01:06.977736874 +0200
@@ -19,7 +19,9 @@
 
 LD=@CXX@
 
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) -DDEBUG_GDS_ALLOC -Wall -Wno-switch -Wno-non-virtual-dtor
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc	2016-08-31 21:03:32.994401873 +0200
@@ -18,6 +18,9 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-PROD_FLAGS=-ggdb -fno-omit-frame-pointer -fsigned-char -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DLINUX -DPowerPC -DPPC -O3
-DEV_FLAGS=-ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -fsigned-char -DLINUX -DPowerPC -DPPC -Wno-non-virtual-dtor
+COMMON_FLAGS=@CFLAGS@ -fsigned-char -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DPowerPC -DPPC
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
+
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 #DEV_FLAGS=-DUSE_VALGRIND -ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -DLINUX -DPowerPC -DPPC -Wno-non-virtual-dtor
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc64.orig	2016-08-31 21:09:42.987730985 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc64	2016-08-31 21:09:45.644397621 +0200
@@ -20,8 +20,8 @@
 
 #LD=@CXX@
 
-#PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-#DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
+COMMON_FLAGS=@CFLAGS@ -DLINUX -DPPC64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
+OPTIMIZE_FLAGS=
 
-PROD_FLAGS=-O3 -DLINUX -DPPC64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DPPC64 -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMON_FLAGS) -Wno-non-virtual-dtor
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc64el.orig	2016-04-14 16:07:29.000000000 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_powerpc64el	2016-08-31 21:10:53.874396836 +0200
@@ -20,8 +20,8 @@
 
 #LD=@CXX@
 
-#PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-#DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
+COMMON_FLAGS=@CFLAGS@ -DLINUX -DPPC64EL -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
+OPTIMIZE_FLAGS=
 
-PROD_FLAGS=-O3 -DLINUX -DPPC64EL -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DPPC64EL -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) -Wno-non-virtual-dtor
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_s390x.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_s390x	2016-08-31 21:05:39.261067098 +0200
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-3.0.0.32483-0/builds/posix/prefix.linux_sparc32.orig	2016-08-31 20:27:07.857760158 +0200
+++ Firebird-3.0.0.32483-0/builds/posix/prefix.linux_sparc32	2016-08-31 21:06:34.644399800 +0200
@@ -18,6 +18,7 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
-PROD_FLAGS=-ggdb -mcpu=ultrasparc -mtune=ultrasparc -O3 $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Wall -Wno-switch  $(COMMON_FLAGS) -Wno-non-virtual-dtor
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
+OPTIMIZE_FLAGS=
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=-p -Wall -Wno-switch  $(COMMON_FLAGS) -Wno-non-virtual-dtor
