--- firebird-1.0.2.908/jrd/val.c.orig	Wed Jun  5 00:23:28 2002
+++ firebird-1.0.2.908/jrd/val.c	Thu Jun 12 14:12:27 2003
@@ -100,8 +100,8 @@
    If other attachments are already made locally or through the same multi-
    client server, validate gives up with the message:
 
-   "Lock timeout during wait transaction
-   -- Object "database_filename.gdb" is in use"
+   <<Lock timeout during wait transaction
+   -- Object "database_filename.gdb" is in use>>
 
    If other processes or servers are attached to the database, validate 
    waits for the exclusive lock on the database (i.e. waits for every 
