--- firebird-1.5.0.4290/builds/posix/Makefile.in.refDatabases.orig	2003-04-10 09:50:03.000000000 +0200
+++ firebird-1.5.0.4290/builds/posix/Makefile.in.refDatabases	2004-05-24 23:46:47.933836000 +0200
@@ -69,7 +69,7 @@
 	-$(RM) $@
 	$(BIN)/create_db $@
 	$(TOUCH) $@
-	$(CHMOD) 444 $@
+	$(CHMOD) 644 $@
 
 # Ok so Jim had/has a sailing theme, and the sample database that a lot
 # of .epp files require to compile is yachts.lnk. MOD 05-Aug-2002
@@ -93,7 +93,7 @@
 	$(RM) -f msg.fdb
 	echo create database \'msg.fdb\'\; | $(ISQL_STATIC)
 	for sn in $(MSG_FILES); do (echo $$sn; $(ISQL_STATIC) -i $$sn msg.fdb) || exit; done
-#	$(CHMOD) 444 msg.fdb
+#	$(CHMOD) 644 msg.fdb
 	$(TOUCH) $@
 
 help.fdb:	$(FIREBIRD)/help/help.fdb
@@ -103,7 +103,7 @@
 $(FIREBIRD)/help/help.fdb:	$(BLD_ROOT)/misc/help.gbak
 	$(BIN)/gbak_static -MODE read_only -R $< $@
 	$(TOUCH) $@
-	$(CHMOD) 444 $@
+	$(CHMOD) 644 $@
 
 $(FIREBIRD)/security2.fdb:	security2.fdb
 	$(RM) -f $@
@@ -118,7 +118,7 @@
 metadata.fdb:	$(BLD_ROOT)/misc/metadata.gbak
 	$(BIN)/gbak_static -MODE read_only -R $< $@
 	$(TOUCH) $@
-	$(CHMOD) 444 $@
+	$(CHMOD) 644 $@
 
 # An alternative metadata creation method, this one is actually preferred
 # since it ensures the data is valid.
@@ -126,7 +126,7 @@
 metadata.fdb.x:	$(SRC_ROOT)/misc/metadata.sql
 	$(BIN)/isql -i $<
 	$(TOUCH) $@
-	$(CHMOD) 444 $@
+	$(CHMOD) 644 $@
 
 
 FORCE:
--- firebird-1.5.0.4290/builds/posix/make.defaults.orig	2003-11-11 02:01:51.000000000 +0100
+++ firebird-1.5.0.4290/builds/posix/make.defaults	2004-05-24 23:47:17.602325704 +0200
@@ -92,9 +92,9 @@
 SH=			sh -c
 RM=			rm -f
 CHMOD=			chmod
-CHMOD_6=		chmod 666
-CHMOD_7=		chmod 777
-CHMOD_S7=		chmod 06777
+CHMOD_6=		chmod 644
+CHMOD_7=		chmod 755
+CHMOD_S7=		chmod 06755
 MV=			mv -f
 TOUCH=			touch
 CP=			cp
