--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux	2012-09-19 19:29:07.407450640 +0200
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -march=i586 -mtune=i686 -fno-omit-frame-pointer
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 
 PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_amd64.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_amd64	2012-09-19 19:29:42.140783248 +0200
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer 
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer 
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 CXXFLAGS:= $(CXXFLAGS) -fno-rtti
 
--- Firebird-2.1.3.18185-0/builds/posix/prefix.linux_arm.orig	2008-04-15 16:11:33.000000000 +0200
+++ Firebird-2.1.3.18185-0/builds/posix/prefix.linux_arm	2010-08-29 10:50:20.766823295 +0200
@@ -23,11 +23,11 @@
 
 #LD=@CXX@
 
-#PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DNDEBUG -DLINUX -pipe -MMD -fPIC
-#DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
+COMMON_FLAGS=@CFLAGS@ -DLINUX -DARM -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
+OPTIMIZE_FLAGS=
 
-PROD_FLAGS=-O3 -DNDEBUG -DLINUX -DARM -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DARM -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) -Wall
 
 OS_ServerFiles=inet_server.cpp
 
--- Firebird-2.1.3.18185-0/builds/posix/prefix.linux_generic.orig	2007-05-03 13:24:50.000000000 +0200
+++ Firebird-2.1.3.18185-0/builds/posix/prefix.linux_generic	2010-08-29 10:50:59.358822457 +0200
@@ -18,10 +18,11 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL
+OPTIMIZE_FLAGS=
 
-PROD_FLAGS=-ggdb -O3 -DNDEBUG $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Wall -Wno-switch $(COMMON_FLAGS)
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=-Wall -Wno-switch $(COMMON_FLAGS)
 CXXFLAGS:= $(CXXFLAGS) -fno-rtti
 
 OS_ServerFiles=inet_server.cpp
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_ia64.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_ia64	2012-09-19 19:34:44.454110281 +0200
@@ -18,8 +18,8 @@
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 # Adriano dos Santos Fernandes - IA-64 port
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DIA64 -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer 
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DIA64 -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer 
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 CXXFLAGS:= $(CXXFLAGS) -fno-rtti
 
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_mips.orig	2012-09-19 19:35:09.124109765 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_mips	2012-09-19 19:35:12.537443029 +0200
@@ -1,5 +1,5 @@
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 
 PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_mipsel.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_mipsel	2012-09-19 19:36:25.827441500 +0200
@@ -19,10 +19,12 @@
 
 LD=@CXX@
 
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC
+OPTIMIZE=-fno-omit-frame-pointer
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DNDEBUG -DLINUX -pipe -MMD -fPIC
-DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE)
+DEV_FLAGS=-DDEBUG_GDS_ALLOC -Wall -Wno-switch $(COMMON_FLAGS)
 
 OS_ServerFiles=inet_server.cpp
 
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_powerpc.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_powerpc	2012-09-19 19:39:04.874104851 +0200
@@ -18,8 +18,11 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-PROD_FLAGS=-ggdb -fno-omit-frame-pointer -fsigned-char -DNDEBUG -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DLINUX -DPowerPC -DPPC -O3
-DEV_FLAGS=-ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -fsigned-char -DLINUX -DPowerPC -DPPC
+COMMON_FLAGS=@CFLAGS@ -fsigned-char -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DPowerPC -DPPC
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
+
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable $(COMMON_FLAGS)
 #DEV_FLAGS=-DUSE_VALGRIND -ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -DLINUX -DPowerPC -DPPC
 
 OS_ServerFiles=inet_server.cpp
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_s390x.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_s390x	2012-09-19 19:39:25.480771086 +0200
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 
 PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
--- Firebird-2.5.1.26351-0/builds/posix/prefix.linux_sparc32.orig	2011-08-12 09:28:48.000000000 +0200
+++ Firebird-2.5.1.26351-0/builds/posix/prefix.linux_sparc32	2012-09-19 19:40:02.704103648 +0200
@@ -18,9 +18,10 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
-PROD_FLAGS=-ggdb -mcpu=ultrasparc -mtune=ultrasparc -O3 -DNDEBUG $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Wall -Wno-switch  $(COMMON_FLAGS)
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
+OPTIMIZE_FLAGS=
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=-p -Wall -Wno-switch  $(COMMON_FLAGS)
 CXXFLAGS:= $(CXXFLAGS) -fno-rtti
 
 OS_ServerFiles=inet_server.cpp
