--- firebird-1.5.0.4290/builds/posix/Makefile.in.refDatabases.orig	2003-04-10 09:50:03.000000000 +0200
+++ firebird-1.5.0.4290/builds/posix/Makefile.in.refDatabases	2004-05-24 23:46:47.933836000 +0200
@@ -107,7 +107,7 @@
 $(FIREBIRD)/security.fdb:	$(SRC_ROOT)/misc/security.gbak
 	$(BIN)/gbak_static  -R $< $@
 	$(TOUCH) $@
-	$(CHMOD) 666 $@
+	$(CHMOD) 644 $@
 
 metadata.fdb:	$(SRC_ROOT)/misc/metadata.gbak
 	$(BIN)/gbak_static -MODE read_only -R $< $@
--- firebird-1.5.0.4290/builds/posix/make.defaults.orig	2003-11-11 02:01:51.000000000 +0100
+++ firebird-1.5.0.4290/builds/posix/make.defaults	2004-05-24 23:47:17.602325704 +0200
@@ -92,9 +92,9 @@
 SH=			sh -c
 RM=			rm -f
 CHMOD=			chmod
-CHMOD_6=		chmod 666
-CHMOD_7=		chmod 777
-CHMOD_S7=		chmod 06777
+CHMOD_6=		chmod 644
+CHMOD_7=		chmod 755
+CHMOD_S7=		chmod 06755
 MV=			mv -f
 TOUCH=			touch
 CP=			cp
