diff -up Firebird-5.0.0.1306-0-source/builds/posix/Makefile.in._orig Firebird-5.0.0.1306-0-source/builds/posix/Makefile.in
--- Firebird-5.0.0.1306-0-source/builds/posix/Makefile.in._orig	2024-02-22 09:14:30.430893458 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/Makefile.in	2024-02-22 09:17:31.106365442 +0100
@@ -495,7 +495,7 @@ SECURITY_TMP = security.tmp
 
 databases:		yachts.lnk $(SECURITY_FDB)
 
-export LD_LIBRARY_PATH := $(LD_LIBRARY_PATH):$(FIREBIRD)/lib
+export LD_LIBRARY_PATH := $(FIREBIRD)/lib:$(LD_LIBRARY_PATH)
 
 yachts.lnk:		metadata.fdb
 	-$(RM) $@
