--- firebird-1.0.2.908/builds/original/prefix.linux.orig	2002-03-01 14:16:31.000000000 +0100
+++ firebird-1.0.2.908/builds/original/prefix.linux	2003-10-29 19:58:41.558232152 +0100
@@ -65,7 +65,7 @@
 BACKEND_BRIDGE_MISC=	head5.o allp.o
 BIN_PATH=		/usr/isc/bin
 BRIDGE_MISC=		head5.o allp.o
-CHMOD_VAL=		666
+CHMOD_VAL=		644
 
 #NOTE: I removed CURSES_LIB=  -lbsd because it breaks the build on SuSE systems
 #      and isn't necessary anymore. 
@@ -163,9 +163,9 @@
 SH=			sh -c
 RM=			rm -f
 CHMOD=			chmod
-CHMOD_6=		chmod 666
-CHMOD_7=		chmod 777
-CHMOD_S7=		chmod 06777
+CHMOD_6=		chmod 644
+CHMOD_7=		chmod 755
+CHMOD_S7=		chmod 06755
 MV=			mv -f
 TOUCH=			touch
 CP=			cp
