--- Firebird-2.5.0.26074-0/src/jrd/common.h.orig	2010-09-08 10:43:27.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/common.h	2010-10-28 07:51:57.395734339 +0200
@@ -158,6 +158,11 @@
 #define RISC_ALIGNMENT
 #endif // IA64
 
+#ifdef __alpha__
+#define IMPLEMENTATION  isc_info_db_impl_linux_alpha	/* 82 */
+#define RISC_ALIGNMENT
+#endif
+
 #ifndef USE_POSIX_THREADS
 // force pthread detection on Linux for distros that do not provide
 // POSIX thread compatability
--- Firebird-2.5.0.26074-0/src/jrd/inf_pub.h.orig	2010-09-08 10:43:27.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/inf_pub.h	2010-10-28 07:52:00.183734339 +0200
@@ -213,6 +213,7 @@
 
 	isc_info_db_impl_linux_sh = 80,
 	isc_info_db_impl_linux_sheb = 81,
+	isc_info_db_impl_linux_alpha = 82,
 
 	isc_info_db_impl_last_value   // Leave this LAST!
 };
--- Firebird-2.5.0.26074-0/src/jrd/pag.cpp.orig	2010-09-08 10:43:27.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/pag.cpp	2010-10-28 07:58:22.771734339 +0200
@@ -160,9 +160,10 @@
 static const int CLASS_LINUX_S390 = 37;		// LINUX/s390
 static const int CLASS_LINUX_SH = 38;		// LINUX/SH (little-endian)
 static const int CLASS_LINUX_SHEB = 39;		// LINUX/SH (big-endian)
+static const int CLASS_LINUX_ALPHA = 40;	// LINUX/Alpha
 
 static const int CLASS_MAX10 = CLASS_LINUX_AMD64;	// This should not be changed, no new ports with ODS10
-static const int CLASS_MAX = CLASS_LINUX_SHEB;
+static const int CLASS_MAX = CLASS_LINUX_ALPHA;
 
 // ARCHITECTURE COMPATIBILITY CLASSES
 
@@ -259,7 +260,8 @@
 	archBigEndian,	  // CLASS_LINUX_S390X
 	archBigEndian,	  // CLASS_LINUX_S390
 	archLittleEndian, // CLASS_LINUX_SH
-	archBigEndian     // CLASS_LINUX_SHEB
+	archBigEndian,    // CLASS_LINUX_SHEB
+	archLittleEndian  // CLASS_LINUX_ALPHA
 };
 
 #ifdef __sun
@@ -319,6 +321,8 @@
 const SSHORT CLASS		= CLASS_LINUX_SH;
 #elif defined(SHEB)
 const SSHORT CLASS		= CLASS_LINUX_SHEB;
+#elif defined(__alpha__)
+const SSHORT CLASS		= CLASS_LINUX_ALPHA;
 #else
 #error no support on other hardware for Linux
 #endif
--- Firebird-2.5.0.26074-0/src/jrd/utl.cpp.orig	2010-09-08 10:43:27.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/utl.cpp	2010-10-28 07:57:59.231736853 +0200
@@ -225,7 +225,8 @@
 	"Firebird/linux s390x",			// 78
 	"Firebird/linux s390",			// 79
 	"Firebird/linux SH",			// 80
-	"Firebird/linux SHEB"			// 81
+	"Firebird/linux SHEB",			// 81
+	"Firebird/linux Alpha"			// 82
 };
 
 
