--- firebird-1.0.0.796/jrd/rse.c.orig	Fri Mar  1 12:54:03 2002
+++ firebird-1.0.0.796/jrd/rse.c	Thu Dec  5 00:48:00 2002
@@ -735,7 +735,7 @@
 		MOV_get_int64 (EVL_expr (tdbb, (NOD) rsb->rsb_arg [0]), 0);
 
             if (((IRSB_FIRST)impure)->irsb_count < 1)
-                ERR_post (gds__bad_limit_param, 0);
+                ERR_post (gds__not_valid /*gds__bad_limit_param*/, 0);
 
 	    rsb = rsb->rsb_next;
 	    break;
@@ -745,7 +745,7 @@
 		MOV_get_int64 (EVL_expr (tdbb, (NOD) rsb->rsb_arg [0]), 0);
 
             if (((IRSB_SKIP)impure)->irsb_count < 0)
-                ERR_post (gds__bad_skip_param, 0);
+                ERR_post (gds__not_valid /*gds__bad_skip_param*/, 0);
             ((IRSB_SKIP)impure)->irsb_count++;
 
 	    rsb = rsb->rsb_next;
--- firebird-1.0.0.796/Configure_SetupEnv.sh.orig	Tue Aug  7 20:19:19 2001
+++ firebird-1.0.0.796/Configure_SetupEnv.sh	Thu Dec  5 08:14:29 2002
@@ -53,6 +53,6 @@
 LD_LIBRARY_PATH=$FBNewBuild/lib:$FBBuildRoot/jrd:$LD_LIBRARY_PATH
 export LD_LIBRARY_PATH
 
-PATH=$PATH:$FBNewBuild/bin:.
+PATH=$FBNewBuild/bin:$PATH:.
 
 
