--- firebird-1.5.0.4290/configure.in.orig	2003-12-23 23:23:54.000000000 +0000
+++ firebird-1.5.0.4290/configure.in	2004-05-28 16:18:05.000000000 +0000
@@ -51,15 +51,6 @@
     SHRLIB_EXT=so
     ;;
 
-  sparc*-*-linux-*)
-    MAKEFILE_PREFIX=linux_sparc32
-    PLATFORM=LINUX
-    AC_DEFINE(LINUX)
-    LOCK_MANAGER_FLG=Y
-    EDITLINE_FLG=Y
-    SHRLIB_EXT=so
-    ;;
-
   *-*-linux*)
     MAKEFILE_PREFIX=linux
     PLATFORM=LINUX
--- firebird-1.5.0.4290/src/jrd/common.h.orig	2003-10-27 18:30:37.000000000 +0000
+++ firebird-1.5.0.4290/src/jrd/common.h	2004-05-28 16:29:42.000000000 +0000
@@ -120,7 +120,7 @@
 #define IMPLEMENTATION  isc_info_db_impl_i386 /* 60  next higher unique number, See you later  */
 #endif /* i386 */
 
-#ifdef sparc
+#ifdef __sparc__
 #define IMPLEMENTATION  isc_info_db_impl_linux_sparc /* 65  */
 #endif /* sparc */
 
--- firebird-1.5.0.4290/src/jrd/pag.cpp.orig	2003-10-08 13:03:06.000000000 +0000
+++ firebird-1.5.0.4290/src/jrd/pag.cpp	2004-05-28 17:14:57.000000000 +0000
@@ -177,7 +177,7 @@
 #ifdef i586
 #define CLASS           19
 #endif
-#ifdef sparc
+#ifdef __sparc__
 #define CLASS           20
 #endif
 #endif
--- firebird-1.5.0.4290/src/remote/xdr.cpp.orig	2003-11-09 19:58:26.000000000 +0000
+++ firebird-1.5.0.4290/src/remote/xdr.cpp	2004-05-28 17:28:21.000000000 +0000
@@ -42,7 +42,7 @@
 #if defined(i386) || defined(I386) || defined(_M_IX86)
 #define		SWAP_DOUBLE
 #else
-#if !defined(sparc) && !defined(PowerPC)
+#if !defined(sparc) && !defined(__sparc__) && !defined(PowerPC)
 #error "Define SWAP_DOUBLE for your platform correctly !"
 #endif
 #endif
