diff -uNr firebird-1.5.6.5026.orig/src/include/fb_types.h firebird-1.5.6.5026/src/include/fb_types.h
--- firebird-1.5.6.5026.orig/src/include/fb_types.h	2004-09-18 01:34:05.000000000 +0200
+++ firebird-1.5.6.5026/src/include/fb_types.h	2012-04-14 22:39:15.973816920 +0200
@@ -74,7 +74,7 @@
 // Temporarly restrict new definition until ULONG clash with Windows
 // type is solved. Win64 port is not possible before that point.
 // Cannot use SIZEOF_LONG define here because we are in a public header
-#if defined(_LP64) || defined(__LP64__) || defined(__arch64__)
+#if defined(_LP64) || defined(__LP64__) || defined(__arch64__) || defined(__alpha__)
 	/* EKU: Firebird requires (S)LONG to be 32 bit */
 #   define LONG_DEFINED
     typedef int SLONG;
diff -uNr firebird-1.5.6.5026.orig/src/jrd/btr.cpp firebird-1.5.6.5026/src/jrd/btr.cpp
--- firebird-1.5.6.5026.orig/src/jrd/btr.cpp	2006-11-05 15:38:08.000000000 +0100
+++ firebird-1.5.6.5026/src/jrd/btr.cpp	2012-04-14 22:39:15.973816920 +0200
@@ -190,7 +190,7 @@
 //
 inline SLONG BTR_get_quad(const UCHAR* p)
 {
-#if defined(i386) || defined(I386) || defined(_M_IX86) || defined(AMD64)
+#if defined(i386) || defined(I386) || defined(_M_IX86) || defined(AMD64) || defined(__alpha__)
 	// For IA32 (little-endian) this optimization is a _very_ large speed-up!
 	return *reinterpret_cast<const SLONG*>(p);
 #else
diff -uNr firebird-1.5.6.5026.orig/src/jrd/common.h firebird-1.5.6.5026/src/jrd/common.h
--- firebird-1.5.6.5026.orig/src/jrd/common.h	2007-12-17 15:48:36.000000000 +0100
+++ firebird-1.5.6.5026/src/jrd/common.h	2012-04-14 22:39:15.953817208 +0200
@@ -132,6 +132,14 @@
 #define IMPLEMENTATION  isc_info_db_impl_linux_sparc /* 65  */
 #endif /* sparc */
 
+#ifdef __powerpc__
+#define IMPLEMENTATION	isc_info_db_impl_linux_ppc
+#endif
+
+#ifdef __alpha__
+#define IMPLEMENTATION	isc_info_db_impl_linux_alpha
+#endif
+
 #define MEMMOVE(from,to,length)		memmove ((void *)to, (void *)from, (size_t) length)
 #define MOVE_FAST(from,to,length)       memcpy (to, from, (int) (length))
 #define MOVE_FASTER(from,to,length)     memcpy (to, from, (int) (length))
diff -uNr firebird-1.5.6.5026.orig/src/jrd/ibase.h firebird-1.5.6.5026/src/jrd/ibase.h
--- firebird-1.5.6.5026.orig/src/jrd/ibase.h	2007-12-17 15:48:37.000000000 +0100
+++ firebird-1.5.6.5026/src/jrd/ibase.h	2012-04-14 22:39:15.957817150 +0200
@@ -57,7 +57,7 @@
 // It is difficult to detect 64-bit long from the redistributable header
 // we do not care of 16-bit platforms anymore thus we may use plain "int"
 // which is 32-bit on all platforms we support
-#if defined(_LP64) || defined(__LP64__) || defined(__arch64__)
+#if defined(_LP64) || defined(__LP64__) || defined(__arch64__) || defined(__alpha__)
 typedef	int		ISC_LONG;
 typedef	unsigned int	ISC_ULONG;
 #else
@@ -1578,6 +1578,9 @@
 	isc_info_db_impl_linux_amd64 = 66,
 	isc_info_db_impl_darwin_x86 = 67,
 
+	isc_info_db_impl_linux_ppc = 68,
+	isc_info_db_impl_linux_alpha = 69,
+
 	isc_info_db_impl_last_value   /* Leave this LAST! */
     };
 
diff -uNr firebird-1.5.6.5026.orig/src/jrd/pag.cpp firebird-1.5.6.5026/src/jrd/pag.cpp
--- firebird-1.5.6.5026.orig/src/jrd/pag.cpp	2008-03-21 09:55:47.000000000 +0100
+++ firebird-1.5.6.5026/src/jrd/pag.cpp	2012-04-14 22:39:15.957817150 +0200
@@ -198,6 +198,12 @@
 #else
 #define CLASS		CLASS_SOLARIS_SPARC
 #endif
+#ifdef __powerpc__
+#define CLASS		30
+#endif
+#ifdef __alpha__
+#define CLASS		31
+#endif
 #endif
 
 #ifdef HPUX
diff -uNr firebird-1.5.6.5026.orig/src/jrd/utl.cpp firebird-1.5.6.5026/src/jrd/utl.cpp
--- firebird-1.5.6.5026.orig/src/jrd/utl.cpp	2007-12-17 15:48:38.000000000 +0100
+++ firebird-1.5.6.5026/src/jrd/utl.cpp	2012-04-14 22:43:20.106303847 +0200
@@ -257,6 +257,9 @@
     "Firebird/linux Sparc",	/* 65 */
     "Firebird/linux AMD64",	/* 66 */
     "Firebird/Darwin/Intel"	/* 67 */
+
+    "Firebird/linux PPC",	/* 68 */
+    "Firebird/linux alpha"	/* 69 */
 };
 
 
diff -uNr firebird-1.5.6.5026.orig/src/remote/xdr.cpp firebird-1.5.6.5026/src/remote/xdr.cpp
--- firebird-1.5.6.5026.orig/src/remote/xdr.cpp	2007-12-17 15:48:42.000000000 +0100
+++ firebird-1.5.6.5026/src/remote/xdr.cpp	2012-04-14 22:46:15.415780168 +0200
@@ -42,7 +42,7 @@
 // The simpliest way to check it is to issue
 // "select abs(2.0/3.0) from rdb$database" from correct client
 // It will return big strange value in case of invalid define
-#if defined(i386) || defined(I386) || defined(_M_IX86) || defined(AMD64)
+#if defined(i386) || defined(I386) || defined(_M_IX86) || defined(AMD64) || defined(__alpha__)
 #define		SWAP_DOUBLE
 #else
 #if !defined(sparc) && !defined(PowerPC) && !defined(HPUX) && !defined(__ppc__)
