--- Firebird-2.5.0.26074-0/builds/posix/make.defaults.orig	2010-10-28 09:05:38.000000000 +0200
+++ Firebird-2.5.0.26074-0/builds/posix/make.defaults	2010-10-28 09:17:10.475739367 +0200
@@ -259,7 +259,7 @@
 else
 LIB_LINK_RPATH = $(call LIB_PLATFORM_RPATH,'$$$$ORIGIN/../$(1)')
 endif
-LIB_PATH_OPTS = $(call LIB_LINK_RPATH,lib) $(call LIB_LINK_RPATH,intl)
+LIB_PATH_OPTS =
 LIB_LINK_SONAME= -Wl,-soname,$(1)
 LIB_LINK_MAPFILE= -Wl,--version-script,
 FBEMBED_LINK= -L$(LIB) -lfbembed
