--- Firebird-4.0.4.3010-0/src/yvalve/gds.cpp.orig	2023-12-20 10:30:04.215137656 +0100
+++ Firebird-4.0.4.3010-0/src/yvalve/gds.cpp	2023-12-20 10:34:30.890122203 +0100
@@ -3950,10 +3950,7 @@
 		if (!fb_utils::readenv(FB_LOCK_ENV, lockPrefix))
 		{
 #if !defined(WIN_NT)
-			const char* tmp = getTemporaryFolder();
-			if (!tmp)
-				tmp = WORKFILE;
-			PathUtils::concatPath(lockPrefix, tmp, LOCKDIR);
+			PathUtils::concatPath(lockPrefix, "/var/lib", LOCKDIR);
 #else
 			char cmnData[MAXPATHLEN];
 			if (SHGetSpecialFolderPath(NULL, cmnData, CSIDL_COMMON_APPDATA, TRUE))
