diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux	2024-02-22 08:47:40.412106924 +0100
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -march=i586 -mtune=i686 -fno-omit-frame-pointer
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing -Wno-unused-local-typedefs
 PLATFORM_PLUSPLUS_FLAGS=-Wno-invalid-offsetof -Wno-class-memaccess
 
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_amd64._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_amd64
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_amd64._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_amd64	2024-02-22 08:49:50.947915307 +0100
@@ -18,8 +18,8 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DAMD64 -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
 WARN_FLAGS=-Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing -Wno-unused-local-typedefs
 PLATFORM_PLUSPLUS_FLAGS=-Wno-invalid-offsetof -Wno-class-memaccess
 #VALGRIND_FLAGS=-DUSE_VALGRIND
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm	2024-02-22 08:51:24.290642044 +0100
@@ -20,7 +20,7 @@
 
 #LD=@CXX@
 
-COMMON_FLAGS=-DLINUX -DARM -pipe -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
+COMMON_FLAGS=@CFLAGS@ -DLINUX -DARM -pipe -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
 
-PROD_FLAGS=$(COMMON_FLAGS) -O3
-DEV_FLAGS=$(COMMON_FLAGS) -p -ggdb -Werror=delete-incomplete -Wall -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) -Werror=delete-incomplete -Wall -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm64._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm64
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm64._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_arm64	2024-02-22 09:08:58.709795676 +0100
@@ -23,5 +23,5 @@
 #PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
 #DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch
 
-PROD_FLAGS=-O3 -DLINUX -DARM64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DARM64 -pipe -p -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
+PROD_FLAGS=@CFLAGS@ -DLINUX -DARM64 -pipe -MMD -fPIC -fsigned-char -fmessage-length=0
+DEV_FLAGS=@CFLAGS@ -DLINUX -DARM64 -pipe -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_generic._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_generic
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_generic._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_generic	2024-02-22 08:54:40.156050585 +0100
@@ -18,7 +18,7 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -DFB_SEND_FLAGS=MSG_NOSIGNAL
 
-PROD_FLAGS=-ggdb -O3 $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Werror=delete-incomplete -Wall -Wno-switch $(COMMON_FLAGS) -Wno-non-virtual-dtor
+PROD_FLAGS=$(COMMON_FLAGS)
+DEV_FLAGS=-Werror=delete-incomplete -Wall -Wno-switch $(COMMON_FLAGS) -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mips._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mips
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mips._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mips	2024-02-22 08:57:58.424615561 +0100
@@ -1,7 +1,7 @@
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
 #DEV_FLAGS=-DUSE_VALGRIND -p $(COMMON_FLAGS) $(WARN_FLAGS)
-DEV_FLAGS=-p $(COMMON_FLAGS) $(WARN_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) $(WARN_FLAGS)
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mipsel._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mipsel
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mipsel._orig	2024-02-22 08:37:49.342919440 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_mipsel	2024-02-22 08:59:04.557427310 +0100
@@ -21,5 +21,5 @@ LD=@CXX@
 
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch -Wno-non-virtual-dtor
+PROD_FLAGS=@CFLAGS@ -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
+DEV_FLAGS=@CFLAGS@ -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc._orig	2024-02-22 08:37:49.346252747 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc	2024-02-22 09:00:37.110050069 +0100
@@ -18,6 +18,6 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-PROD_FLAGS=-ggdb -fno-omit-frame-pointer -fsigned-char -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DLINUX -DPowerPC -DPPC -O3
-DEV_FLAGS=-ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -fsigned-char -DLINUX -DPowerPC -DPPC -Wno-non-virtual-dtor
+PROD_FLAGS=@CFLAGS@ -fno-omit-frame-pointer -fsigned-char -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -DLINUX -DPowerPC -DPPC -O3
+DEV_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -fsigned-char -DLINUX -DPowerPC -DPPC -Wno-non-virtual-dtor
 #DEV_FLAGS=-DUSE_VALGRIND -ggdb -O0 -DLINUX -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0 -DLINUX -DPowerPC -DPPC -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64._orig	2024-02-22 08:37:49.346252747 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64	2024-02-22 09:01:39.009585755 +0100
@@ -23,5 +23,5 @@
 #PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
 #DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch
 
-PROD_FLAGS=-O3 -DLINUX -DPPC64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DPPC64 -pipe -p -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
+PROD_FLAGS=@CFLAGS@ -DLINUX -DPPC64 -pipe -MMD -fPIC -fsigned-char -fmessage-length=0
+DEV_FLAGS=@CFLAGS@ -DLINUX -DPPC64 -pipe -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64el._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64el
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64el._orig	2024-02-22 08:37:49.346252747 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_powerpc64el	2024-02-22 09:02:33.409184029 +0100
@@ -23,5 +23,5 @@
 #PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
 #DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Werror=delete-incomplete -Wall -Wno-switch
 
-PROD_FLAGS=-O3 -DLINUX -DPPC64EL -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0
-DEV_FLAGS=-ggdb -DLINUX -DPPC64EL -pipe -p -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
+PROD_FLAGS=@CFLAGS@ -DLINUX -DPPC64EL -pipe -MMD -fPIC -fsigned-char -fmessage-length=0
+DEV_FLAGS=@CFLAGS@ -DLINUX -DPPC64EL -pipe -MMD -fPIC -Werror=delete-incomplete -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_s390x._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_s390x
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_s390x._orig	2024-02-22 08:37:49.346252747 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_s390x	2024-02-22 09:03:36.152060506 +0100
@@ -18,10 +18,10 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char -Wno-invalid-offsetof
-OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=@CFLAGS@ -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0 -fsigned-char -Wno-invalid-offsetof
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer -fno-builtin
 WARN_FLAGS=-Werror=delete-incomplete -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-non-virtual-dtor
 
 PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
 #DEV_FLAGS=-DUSE_VALGRIND -p $(COMMON_FLAGS) $(WARN_FLAGS)
-DEV_FLAGS=-p $(COMMON_FLAGS) $(WARN_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) $(WARN_FLAGS)
diff -up Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_sparc32._orig Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_sparc32
--- Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_sparc32._orig	2024-02-22 08:37:49.346252747 +0100
+++ Firebird-5.0.0.1306-0-source/builds/posix/prefix.linux_sparc32	2024-02-22 09:05:08.354733545 +0100
@@ -18,6 +18,6 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major cleanup
 
-COMMON_FLAGS=-DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
-PROD_FLAGS=-ggdb -mcpu=ultrasparc -mtune=ultrasparc -O3 $(COMMON_FLAGS)
-DEV_FLAGS=-ggdb -p -Werror=delete-incomplete -Wall -Wno-switch  $(COMMON_FLAGS) -Wno-non-virtual-dtor
+COMMON_FLAGS=@CFLAGS@ -DLINUX -pipe -MMD -fPIC -Dsparc -DFB_SEND_FLAGS=MSG_NOSIGNAL
+PROD_FLAGS=$(COMMON_FLAGS)
+DEV_FLAGS=@CFLAGS@ -Werror=delete-incomplete -Wall -Wno-switch  $(COMMON_FLAGS) -Wno-non-virtual-dtor
