diff -urNp -x '*.orig' Firebird-3.0.7.33374-0.org/builds/posix/make.defaults Firebird-3.0.7.33374-0/builds/posix/make.defaults
--- Firebird-3.0.7.33374-0.org/builds/posix/make.defaults	2020-10-20 08:40:05.000000000 +0000
+++ Firebird-3.0.7.33374-0/builds/posix/make.defaults	2022-04-04 17:50:57.999465936 +0000
@@ -142,9 +142,9 @@ SH=			sh -c
 RM=			rm -f
 RM_R=		rm -rf
 CHMOD=			chmod
-CHMOD_6=		chmod 666
-CHMOD_7=		chmod 777
-CHMOD_S7=		chmod 06777
+CHMOD_6=		chmod 644
+CHMOD_7=		chmod 755
+CHMOD_S7=		chmod 06755
 MV=			mv -f
 TOUCH=			touch
 CP=			cp
