--- Firebird-3.0.0.32483-0/builds/install/arch-specific/linux/makeInstallImage.sh.in.orig	2016-04-14 16:07:29.000000000 +0200
+++ Firebird-3.0.0.32483-0/builds/install/arch-specific/linux/makeInstallImage.sh.in	2016-08-31 21:15:58.897726690 +0200
@@ -28,11 +28,6 @@
 # This script builds an image of the installed system into
 # the gen/buildroot directory.
 
-if [ "`whoami`" != "root" ]; then
-	echo 'You must be root to build package'
-	exit 1
-fi
-
 # Making an assumption that this program is being run in the gen directory
 BuildRootDir=..
 BuiltFBDir=Release/firebird  # Where the just build fb exists.
