--- Firebird-2.1.1.17910-0/src/jrd/gds.cpp.orig	2008-01-11 13:56:56.000000000 +0100
+++ Firebird-2.1.1.17910-0/src/jrd/gds.cpp	2008-07-18 22:34:52.502041412 +0200
@@ -2816,7 +2816,9 @@
 	va_list args;
 
 	va_start(args, string);
-	blr_format(control, string, args);
+	Firebird::string temp;
+	temp.vprintf(string, ptr);
+	control->ctl_string += temp;
 	va_end(args);
 	offset = 0;
 	blr_print_line(control, (SSHORT) offset);
