--- Firebird-3.0.0.32483-0/src/yvalve/gds.cpp.orig	2016-08-30 21:31:23.418702669 +0200
+++ Firebird-3.0.0.32483-0/src/yvalve/gds.cpp	2016-08-31 20:24:57.044428320 +0200
@@ -3785,7 +3785,7 @@
 		if (!fb_utils::readenv(FB_LOCK_ENV, lockPrefix))
 		{
 #ifndef WIN_NT
-			PathUtils::concatPath(lockPrefix, WORKFILE, LOCKDIR);
+			PathUtils::concatPath(lockPrefix, "/var/lib", LOCKDIR);
 #else
 #ifdef WIN9X_SUPPORT
 			// shell32.dll version 5.0 and later supports SHGetFolderPath entry point
