--- Firebird-2.5.0.26074-0/src/gpre/pas.cpp.orig	2010-09-08 10:43:39.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/gpre/pas.cpp	2010-10-28 08:44:04.123740205 +0200
@@ -1800,7 +1800,7 @@
 	PAT args;
 	args.pat_database = (gpre_dbb*) init->nod_arg[3];
 	args.pat_vector1 = status_vector(action);
-	args.pat_value1 = (int) init->nod_arg[2];
+	args.pat_value1 = (int) (IPTR)init->nod_arg[2];
 	args.pat_value2 = (int) event_list->nod_count;
 	args.pat_string1 = GDS_EVENT_WAIT;
 	args.pat_string2 = GDS_EVENT_COUNTS;
@@ -1872,7 +1872,7 @@
 		gpre_sym* stack_name = (gpre_sym*) event_init->nod_arg[0];
 		if (!strcmp(event_name->sym_string, stack_name->sym_string))
 		{
-			ident = (int) event_init->nod_arg[2];
+			ident = (int) (IPTR)event_init->nod_arg[2];
 			database = (gpre_dbb*) event_init->nod_arg[3];
 		}
 	}
