--- Firebird-2.5.0.26074-0/src/jrd/gds.cpp.orig	2010-10-28 08:01:51.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/gds.cpp	2010-10-28 08:33:54.483737691 +0200
@@ -1751,7 +1751,12 @@
 
 	GDS_init_prefix();
 
-	strcpy(string, fb_prefix_lock);	// safe - no BO
+	const char *lockdir = "/var/lib/firebird";
+	struct stat sb;
+	if (stat(lockdir, &sb) == 0 && (sb.st_mode & S_IFMT) == S_IFDIR)
+		strcpy(string, lockdir);
+	else
+		strcpy(string, fb_prefix_lock);	// safe - no BO
 
 	// if someone wants to know prefix for lock files,
 	// sooner of all he wants that directory to exist
--- Firebird-2.5.0.26074-0/src/jrd/os/config_root.h.orig	2010-09-08 10:43:20.000000000 +0200
+++ Firebird-2.5.0.26074-0/src/jrd/os/config_root.h	2010-10-28 08:36:33.139734339 +0200
@@ -91,7 +91,7 @@
 	{
 		GetInstallDir();
 		GetRoot();
-		config_file = root_dir + string(CONFIG_FILE);
+		config_file = "/etc/firebird/" + string(CONFIG_FILE);
 	}
 
 	virtual ~ConfigRoot() {}
diff -U 3 -H -d -r -N -- Firebird-2.1.1.17910-0~/src/utilities/ibmgr/srvrmgr.cpp Firebird-2.1.1.17910-0/src/utilities/ibmgr/srvrmgr.cpp
--- Firebird-2.1.1.17910-0~/src/utilities/ibmgr/srvrmgr.cpp	2007-03-28 06:20:36.000000000 +0200
+++ Firebird-2.1.1.17910-0/src/utilities/ibmgr/srvrmgr.cpp	2008-08-23 16:16:02.000000000 +0200
@@ -272,7 +272,7 @@
 		rs = "can not quit now, use shut -ign";
 		break;
 	case MSG_STARTERR:
-		rs = "check $FIREBIRD/firebird.log file for errors";
+		rs = "check /var/log/firebird.log file for errors";
 		break;
 	case MSG_STARTFAIL:
 		rs = "can not start server";
