--- Firebird-2.1.1.17910-0/configure.in.orig	2008-05-27 11:26:28.000000000 +0200
+++ Firebird-2.1.1.17910-0/configure.in	2008-07-18 22:28:11.790034707 +0200
@@ -566,10 +566,10 @@
 dnl AC_CHECK_LIB(c, main)
 AC_CHECK_LIB(m, main)
 if test "$EDITLINE_FLG" = "Y"; then
+	AC_CHECK_LIB(tinfo, tgetent, TERMLIB=tinfo, \
 	AC_CHECK_LIB(curses, tgetent, TERMLIB=curses, \
 		AC_CHECK_LIB(ncurses, tgetent, TERMLIB=ncurses, \
 			AC_CHECK_LIB(termcap, tgetent, TERMLIB=termcap, \
-				AC_CHECK_LIB(tinfo, tgetent, TERMLIB=tinfo, \
 					AC_MSG_ERROR(termcap support not found)))))  
 	AC_SUBST(TERMLIB)
 	XE_APPEND(-l$TERMLIB, LIBS)
