Loosen a bit a module filename check

diff -up Firebird-5.0.0.1306-0-source/src/common/os/posix/mod_loader.cpp._orig Firebird-5.0.0.1306-0-source/src/common/os/posix/mod_loader.cpp
--- Firebird-5.0.0.1306-0-source/src/common/os/posix/mod_loader.cpp._orig	2024-02-22 09:31:12.670991187 +0100
+++ Firebird-5.0.0.1306-0-source/src/common/os/posix/mod_loader.cpp	2024-02-22 09:34:31.349588488 +0100
@@ -199,7 +199,6 @@ void* DlfcnModule::findSymbol(ISC_STATUS
 		symbolPath = info.dli_fname;
 
 	const char* errText = "Actual module name does not match requested";
-	if (PathUtils::isRelative(libraryPath) || PathUtils::isRelative(symbolPath))
 	{
 		// check only name (not path) of the library
 		Firebird::PathName dummyDir, nm1, nm2;
@@ -211,11 +210,6 @@ void* DlfcnModule::findSymbol(ISC_STATUS
 			return NULL;
 		}
 	}
-	else if (libraryPath != symbolPath)
-	{
-		makeErrorStatus(status, errText);
-		return NULL;
-	}
 #endif
 
 	return result;
