Loosen a bit a module filename check

diff -up Firebird-3.0.10.33601-0/src/common/os/posix/mod_loader.cpp._orig Firebird-3.0.10.33601-0/src/common/os/posix/mod_loader.cpp
--- Firebird-3.0.10.33601-0/src/common/os/posix/mod_loader.cpp._orig	2022-06-07 10:18:52.000000000 +0200
+++ Firebird-3.0.10.33601-0/src/common/os/posix/mod_loader.cpp	2022-06-30 14:14:59.963868203 +0200
@@ -149,7 +149,6 @@ void* DlfcnModule::findSymbol(const Fire
 	if (!dladdr(result, &info))
 		return NULL;
 
-	if (PathUtils::isRelative(fileName) || PathUtils::isRelative(info.dli_fname))
 	{
 		// check only name (not path) of the library
 		Firebird::PathName dummyDir, nm1, nm2;
@@ -158,8 +157,6 @@ void* DlfcnModule::findSymbol(const Fire
 		if (nm1 != nm2)
 			return NULL;
 	}
-	else if (fileName != info.dli_fname)
-		return NULL;
 #endif
 
 	return result;
