Loosen a bit a module filename check

--- Firebird-3.0.11.33703-0/src/common/os/posix/mod_loader.cpp.orig	2023-09-29 13:31:01.412025778 +0200
+++ Firebird-3.0.11.33703-0/src/common/os/posix/mod_loader.cpp	2023-09-29 14:06:07.234937115 +0200
@@ -166,7 +166,6 @@
 	if (!realpath(info.dli_fname, symbolPathBuffer))
 		symbolPath = info.dli_fname;
 
-	if (PathUtils::isRelative(libraryPath) || PathUtils::isRelative(symbolPath))
 	{
 		// check only name (not path) of the library
 		Firebird::PathName dummyDir, nm1, nm2;
@@ -175,8 +174,6 @@
 		if (nm1 != nm2)
 			return NULL;
 	}
-	else if (libraryPath != symbolPath)
-		return NULL;
 #endif
 
 	return result;
