Loosen a bit a module filename check

diff -up Firebird-4.0.4.3010-0/src/common/os/posix/mod_loader.cpp._orig Firebird-4.0.4.3010-0/src/common/os/posix/mod_loader.cpp
--- Firebird-4.0.4.3010-0/src/common/os/posix/mod_loader.cpp._orig	2023-11-08 13:07:47.000000000 +0100
+++ Firebird-4.0.4.3010-0/src/common/os/posix/mod_loader.cpp	2023-12-22 13:56:00.026703313 +0100
@@ -184,7 +184,6 @@ void* DlfcnModule::findSymbol(ISC_STATUS
 	}
 
 	const char* errText = "Actual module name does not match requested";
-	if (PathUtils::isRelative(fileName) || PathUtils::isRelative(info.dli_fname))
 	{
 		// check only name (not path) of the library
 		Firebird::PathName dummyDir, nm1, nm2;
@@ -196,11 +195,6 @@ void* DlfcnModule::findSymbol(ISC_STATUS
 			return NULL;
 		}
 	}
-	else if (fileName != info.dli_fname)
-	{
-		makeErrorStatus(status, errText);
-		return NULL;
-	}
 #endif
 
 	return result;
